lib/LeagueUtil.js

```diff
@@ -0,0 +1,294 @@
+const deepmerge = require('deepmerge');
+const fs = require('fs');
+const path = require('path');
+
+const ParameterError = require('./errors/ParameterError');
+
+const RateLimiter = require('./RateLimiter');
+
+class LeagueUtil {
+	// TODO: maybe split into more specific classes instead of "Util" class where everything gets dumped into
```

Comment:
Not all utility functionality will be clearly assignable to an endpoint (and most of them would then again end up within the statc-data util I guess).
So I think a structure independent from the endpoints would be cleaner and easier to discover.

Methods specific for the endpoints still can be added to the endpoints for easier use, but internally use and extend the raw util functionalities (which all should be implemented as static and pure functions).